src/dodal/beamlines/i22.py

```diff
@@ -27,30 +31,41 @@
 set_log_beamline(BL)
 set_utils_beamline(BL)
 
+IS_LAB = BL == "p38"
+
 # Currently we must hard-code the visit, determining the visit at runtime requires
 # infrastructure that is still WIP.
 # Communication with GDA is also WIP so for now we determine an arbitrary scan number
 # locally and write the commissioning directory. The scan number is not guaranteed to
 # be unique and the data is at risk - this configuration is for testing only.
-set_directory_provider(
+directory_provider = (
     StaticVisitDirectoryProvider(
+        BL,
+        Path("/dls/p38/data/2024/cm37282-2/bluesky"),
+        client=LocalDirectoryServiceClient(),
+    )
+    if IS_LAB
+    else StaticVisitDirectoryProvider(
         BL,
         Path("/dls/i22/data/2024/cm37271-2/bluesky"),
         client=DirectoryServiceClient("http://i22-control:8088/api"),
     )
 )
 
+set_directory_provider(directory_provider)
 
+
+# d11 at p38, but disconnected
 def saxs(
     wait_for_connection: bool = True, fake_with_ophyd_sim: bool = False
 ) -> PilatusDetector:
     return device_instantiation(
-        NXSasPilatus,
-        "saxs",
-        "-EA-PILAT-01:",
+        AravisDetector if IS_LAB else NXSasPilatus,
```

Comment:
I think because they're returning different types, and the metadataholder (while a temporary construct until we have metadata service) would have completely different information it's clearer to have the obvious branch.

`saxs` should also have a return type of either `PilatusDetector | AravisDetector` or `StandardDetector` (see also for the other device factory functions)

Reply:
I think we want the device name to be the same too?

When the device_instantiation is refactored it'll come from the function name, so it's clearer to enforce that now (if this is how we unify p38/i22)

Reply:
I prefer the narrow OR )`|` than the broad `StandardDetector`